org.adempiere.base/src/org/compiere/model/MCostDetail.java

```diff
@@ -864,6 +1148,37 @@ public boolean isDelta()
 			&& getDeltaQty().signum() == 0);
 	}	//	isDelta
 
+	
+	@Override
+	protected boolean beforeSave(boolean newRecord) {
+		if (newRecord) {
+			final String sql = "SELECT MAX(DateAcct) FROM M_CostDetail WHERE M_Product_ID=? AND Processed='Y'";
+			Timestamp MaxDateAcct = DB.getSQLValueTS(get_TrxName(), sql, getM_Product_ID());
+			Timestamp today = TimeUtil.getDay(System.currentTimeMillis());
+			if (MaxDateAcct != null && MaxDateAcct.after(today))
+				today = MaxDateAcct;
```

Comment:
why is MaxDateAcct needed here ? If it is for IsBackDate logic then it should only be use for that and not as default for DateAcct column.

Reply:
Yes, the problem is MaxDateAcct is use as default for dateAcct when dateAcct is null MaxDateAcct > today.

Reply:
Fixed. The DateAcct should defaulted to today instead of MaxDateAcct.